### PR TITLE
Adding helper function for common tests: ping http tcp udp

### PR DESCRIPTION
**Pull request by clu2xlu**

### CODE REVIEW

File: test-e2e/helper.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        logger.info("Test: {} can ping {}".format(ep2.ip, ep1.ip))
+        if ep2.do_ping(ep1.ip):
+            logger.info("Ping test from {} to {} was Successfull".format(ep2.name, ep1.name))
+        else:
+            logger.info("Ping test from {} to {} was Successfull".format(ep2.name, ep1.name))
+
+def do_http_test(test, ep1, ep2):
+    logger.info("-------Test: HTTP test-------")
+    #ep1.do_httpd()
+    #ep2.do_httpd()
```
Comment (zasherif) [inline test-e2e/helper.py:28]: remove these?

File: test-e2e/helper.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    if ep1.do_curl("http://{}:8000 -Ss -m 1".format(ep2.ip)):
+        logger.info("Http test from {} to {} was Successfull".format(ep1.name, ep2.name))
+    else:
+        logger.info("Http test from {} to {} FAILED".format(ep1.name, ep2.name))
+
+def do_tcp_test(test, ep1, ep2):
+    logger.info("-------Test: TCP test-------")
+    #ep1.do_tcp_serve()
+    #ep2.do_tcp_serve()
```
Comment (zasherif) [inline test-e2e/helper.py:43]: ditto

File: test-e2e/helper.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    if ep1.do_tcp_client(ep2.ip):
+        logger.info("TCP test from {} to {} was Successfull".format(ep2.name, ep1.name))
+    else:
+        logger.info("TCP test from {} to {} FAILED".format(ep2.name, ep1.name))
+
+def do_udp_test(test, ep1, ep2):
+    logger.info("-------Test: UDP test-------")
+    #ep1.do_udp_serve()
+    #ep2.do_udp_serve()
```
Comment (zasherif) [inline test-e2e/helper.py:58]: ditto

File: test-e2e/k8spod.py [hunk 4]
```diff
@@ -7,10 +8,82 @@ def __init__(self, api, name, ip):
...
+    #     if err['status'] != "Success":
+    #         return False
+    #     return True
+
+    # def do_udp_serve(self):
+    #     cmd = "nc -u -l -p 5001 > /tmp/udp_recv_file"
+    #     err = self.api.pod_exec(self.name, cmd)
+    #     if err['status'] != "Success":
+    #         return False
+    #     return True
```
Comment (zasherif) [inline test-e2e/k8spod.py:47]: ditto

File: test-e2e/k8spod.py [hunk 5]
```diff
@@ -7,10 +8,82 @@ def __init__(self, api, name, ip):
...
+
+    def do_tcp_client(self, ip, bs=100, count=1, interval=0, detach=False):
+        delay = ''
+        if interval > 0:
+            delay = '-i{}'.format(interval)
+        #dd_cmd = 'dd if=/dev/urandom  bs={} count={} | tee /tmp/tcp_sent_file'.format(bs, count)
+        #cmd = '/bin/sh -c ({} | nc {} 9001 {} -w1)'.format(dd_cmd, ip, delay)
```
Comment (zasherif) [inline test-e2e/k8spod.py:54]: ditto

File: test-e2e/k8spod.py [hunk 6]
```diff
@@ -7,10 +8,82 @@ def __init__(self, api, name, ip):
...
+        if err['status'] != "Success":
+            return False
+        return True
+
+    def do_udp_client(self, ip, bs=100, count=1, interval=0, detach=False):
+        delay = ''
+        if interval > 0:
+            delay = '-i{}'.format(interval)
+        #dd_cmd = 'dd if=/dev/urandom  bs={} count={} | tee /tmp/udp_sent_file'.format(bs, count)
+        #cmd = '/bin/sh -c ({} | nc -u {} 5001 {} -w1)'.format(dd_cmd, ip, delay)
```
Comment (zasherif) [inline test-e2e/k8spod.py:67]: ditto

File: test-e2e/k8spod.py [hunk 7]
```diff
@@ -7,10 +8,82 @@ def __init__(self, api, name, ip):
...
+    #         return False
+    #     return True
+    #
+    # def do_diff_tcp(self, client_ep, server_ep):
+    #     cmd = 'diff /tmp/udp_recv_file /tmp/udp_sent_file'
+    #     err = self.api.pod_exec(self.name, cmd)
+    #     print("DIFF UDP ERROR: {}".format(err))
+    #     if err['status'] != "Success":
+    #         return False
+    #     return True
```
Comment (zasherif) [inline test-e2e/k8spod.py:89]: Why these are commented? How would we know the tcp/udp servers have actually transferred the files?

File: test-e2e/k8spod.py [hunk 8]
```diff
@@ -7,10 +8,82 @@ def __init__(self, api, name, ip):
...
+
+    # def do_diff_tcp(self, client_ep, server_ep):
+    #     cmd = 'diff /tmp/tcp_recv_file /tmp/tcp_sent_file'
+    #     err = self.api.pod_exec(self.name, cmd)
+    #     print("DIFF TCP ERROR: {}".format(err))
+    #     if err['status'] != "Success":
+    #         return False
+    #     return True
+    #
+    # def do_diff_tcp(self, client_ep, server_ep):
```
Comment (zasherif) [inline test-e2e/k8spod.py:83]: do_diff_udp ?

File: test-e2e/helper.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
+import unittest
+import os
+from common import *
+from k8s import *
```
Comment (phudtran) [inline test-e2e/helper.py:4]: We only need to import common here.

File: test-e2e/test_basic_switch.py [hunk 10]
```diff
@@ -3,25 +3,21 @@
 import os
 from common import *
```
Comment (phudtran) [inline test-e2e/test_basic_switch.py:4]: We don 't need to import os and common here.

File: test-e2e/test_basic_switch.py [hunk 11]
```diff
@@ -3,25 +3,21 @@
...
-        self.ep1 = self.api.create_pod("ep6")
-        self.ep2 = self.api.create_pod("ep7")
+        self.ep1 = self.api.create_pod("ep1")
+        self.ep2 = self.api.create_pod("ep2")
 
     def tearDown(self):
-        logger.info("Tearing down the test")
-        self.api.delete_pod("ep6")
-        self.api.delete_pod("ep7")
+        print("Tearing down endpoints....")
```
Comment (phudtran) [inline test-e2e/test_basic_switch.py:18]: Please use logger here.

File: test-e2e/helper.py [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+
+    if ep1.do_curl("http://{}:8000 -Ss -m 1".format(ep2.ip)):
+        logger.info("Http test from {} to {} was Successfull".format(ep1.name, ep2.name))
+    else:
+        logger.info("Http test from {} to {} FAILED".format(ep1.name, ep2.name))
+
+def do_tcp_test(test, ep1, ep2):
+    logger.info("-------Test: TCP test-------")
+
+    #if ep2.do_tcp_client(ep1.ip) and ep2.do_tcp_diff(ep1.name):
```
Comment (phudtran) [inline test-e2e/helper.py:39]: Lets remove this commented out code.

File: test-e2e/helper.py [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    if ep2.do_tcp_client(ep1.ip):
+        logger.info("TCP test from {} to {} was Successfull".format(ep1.name, ep2.name))
+    else:
+        logger.info("TCP test from {} to {} FAILED".format(ep1.name, ep2.name))
+
+    #if ep1.do_tcp_client(ep2.ip) and ep1.do_tcp_diff(ep2.name):
```
Comment (phudtran) [inline test-e2e/helper.py:45]: ditto

File: test-e2e/helper.py [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    if ep1.do_tcp_client(ep2.ip):
+        logger.info("TCP test from {} to {} was Successfull".format(ep2.name, ep1.name))
+    else:
+        logger.info("TCP test from {} to {} FAILED".format(ep2.name, ep1.name))
+
+def do_udp_test(test, ep1, ep2):
+    logger.info("-------Test: UDP test-------")
+
+    #if ep2.do_udp_client(ep1.ip) and ep2.do_udp_diff(ep1.name):
```
Comment (phudtran) [inline test-e2e/helper.py:54]: ditto

File: test-e2e/helper.py [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+    if ep2.do_udp_client(ep1.ip):
+        logger.info("UDP test from {} to {} was Successfull".format(ep1.name, ep2.name))
+    else:
+        logger.info("UDP test from {} to {} FAILED".format(ep1.name, ep2.name))
+
+    #if ep1.do_udp_client(ep2.ip) and ep1.do_udp_diff(ep2.name):
```
Comment (phudtran) [inline test-e2e/helper.py:60]: ditto

**Pull request merged**